### PR TITLE
feat: Zero Presale Integration

### PR DESCRIPTION
**Pull request by morasumu**
# :books: Linear Ticket :books:

- I added the `ZERO` token as one of the default token under RSK chain.
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L50
- added `origins` swap provider
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L90
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L145
- force to add `ZERO` token to the asset list as `@liquality/cryptoassets` doesn't have `ZERO`.
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/store/actions/createWallet.js#L8
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/store/getters.js#L12
- added swap provider under `src/swaps/origins`

### CODE REVIEW

**Comment by remedcu** [comment]
@business-helper Could you please add a general detail of what is being done here: https://github.com/liquality/wallet/pull/408#issue-1061419944

Also, please change the name of the PR to `feat: Zero presale integration`

File: src/store/actions/createWallet.js [hunk 1]
```diff
@@ -2,7 +2,21 @@ import { v4 as uuidv4 } from 'uuid'
 import { encrypt } from '../../utils/crypto'
 import buildConfig from '../../build.config'
 import { accountCreator, getNextAccountColor } from '@/utils/accounts'
-import { ChainId, chains, assets as cryptoassets } from '@liquality/cryptoassets'
+import { ChainId, chains, assets as cryptoassetsOrigin } from '@liquality/cryptoassets'
+
+// TO-DO: remove this after npm package is updated.
```
Comment (bradleySuira) [inline src/store/actions/createWallet.js:7]: I can update the crypto assets package for you let me know, or will need to wait for any other details related to this assets ?

BR

Comment (bradleySuira) [reply src/store/actions/createWallet.js:7]: @business-helper

Comment (business-helper) [reply src/store/actions/createWallet.js:7]: Yes, I think you can update the assets package,
though smart contracts are not deployed yet, you can add ZERO to the assets as the published version doesn't list ZERO, but I will use it in the development.
After smart contract deployment, we can update the assets again with real data.

**Comment by business-helper** [comment]
> @business-helper Could you please add a general detail of what is being done here: [#408 (comment)](https://github.com/liquality/wallet/pull/408#issue-1061419944)
> 
> Also, please change the name of the PR to `feat: Zero presale integration`

sure.
- I added the `ZERO` token as one of the default token under RSK chain.
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L50
- added `origins` swap provider
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L90
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/build.config.js#L145
- force to add `ZERO` token to the asset list as `@liquality/cryptoassets` doesn't have `ZERO`.
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/store/actions/createWallet.js#L8
https://github.com/Sovryn-Origins/wallet/blob/07e0531b3d8dbd0298d9d82d3c25bc45664c1a9f/src/store/getters.js#L12
- added swap provider under `src/swaps/origins`

**Review by bradleySuira — APPROVED** [review]
LGTM

File: src/store/actions/createWallet.js (hunk 1 above)
Comment (bradleySuira) [reply src/store/actions/createWallet.js:7]: @business-helper can you please help me to get some ZERO in testnet ? to test the ui

File: src/swaps/origins/abi/token.json [hunk 2]
```diff
@@ -0,0 +1,310 @@
+[
```
Comment (kraikov) [inline src/swaps/origins/abi/token.json:1]: Can we use
`import ERC20 from '@uniswap/v2-core/build/ERC20.json'` instead?

**Review by kraikov — CHANGES_REQUESTED** [review]

File: src/swaps/origins/abi/token.json (hunk 2 above)
Comment (business-helper) [reply src/swaps/origins/abi/token.json:1]: You're right, I can use ERC20 for the `ZERO` token abi.

File: src/build.config.js [hunk 3]
```diff
@@ -124,6 +141,14 @@ export default {
...
         routerAddressRBTC: SovrynMainnetAddresses.proxy3,
         rpcURL: 'https://public-node.rsk.co/'
+      },
+      origins: {
+        name: 'Origins',
+        icon: 'origins.svg',
+        type: SwapProviderType.ORIGINS,
+        routerAddress: SovrynTestnetAddresses.swapNetwork,
+        // routerAddressRBTC: SovrynTestnetAddresses.proxy3,
+        rpcURL: 'https://public-node.testnet.rsk.co/'
```
[the remark this answers is not in the document]
Comment (business-helper) [reply src/build.config.js:153]: Yes, that's the place to set the mainnet values.

File: src/swaps/origins/OriginsSwapProvider.js [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    const isClosed = await erc20.isClosed()
+    return isClosed
+  }
+
+  async getDepositRate (network) {
+    const erc20 = new ethers.Contract(this.config.presaleAddress, presaleABI, this._getApi(network, 'ZERO'))
+
+    const PPM = (await erc20.PPM()).toString()
+    const exchangeRate = (await erc20.exchangeRate()).toString()
+    return BN(exchangeRate).dividedBy(BN(PPM)).toNumber()
```
Comment (skeremidchiev) [inline src/swaps/origins/OriginsSwapProvider.js:67]: Using `toNumber()` might cause precision loss due to scientific notation.

Comment (business-helper) [reply src/swaps/origins/OriginsSwapProvider.js]: I changed it to `toString()`. thanks!

**Pull request closed without merging**